### ISSUE TITLE
file_util::copy_stream support for copy_file_range on Linux 4.5

### ISSUE THREAD

**Issue by lilianmoraru** [tailhook/vagga#386]
I was reading yesterday about [copy_file_range](http://man7.org/linux/man-pages/man2/copy_file_range.2.html) API extension in Linux 4.5 and first thing I remembered was the stack overflow issue with the big buffer for file copying.

This is a feature request to support `copy_file_range` on Linux 4.5.

**Comment by tailhook** [comment]
Yeah. I'm dreaming about that too. Unfortunately, this only works in the same mountpoint. So we need to make some changes to the layout of the mountpoints in "capsule" (our temporary file system). And we'll be doing that shortly after 0.7.0 released.
